app/controllers/GuidanceController.scala

```diff
@@ -38,16 +40,29 @@ class GuidanceController @Inject() (
   def startJourney(processId: String): Action[AnyContent] = Action.async { implicit request =>
     service.getStartPageUrl(processId).map { url =>
       val relativeUrl = s"/guidance$url"
-      Redirect(relativeUrl).withSession("processId" -> processId)
+      val sessionId: String = hc.sessionId.fold(processId)(_.value)
+      Redirect(relativeUrl).withSession(SessionKeys.sessionId -> sessionId)
     }
   }
 
   def getPage(path: String, questionPageUrl: Option[String] = None): Action[AnyContent] = Action.async { implicit request =>
     val url = "/" + questionPageUrl.fold[String](path)(url => url.drop("/guidance/".length))
-    val processId = request.session.get("processId").get
-    service.getPage(url, processId).map {
-      case Some(page) => Ok(view(page))
-      case _ => NotFound(errorHandler.notFoundTemplate)
+    request.session.get(SessionKeys.sessionId).map{ sessionId =>
+      Logger.info(s"getPage: sessionId = $sessionId")
```

Comment:
It is good to see a logger in use, but as Mac showed me yesterday it is good to create an instance of the logger as in

val logger: Logger = Logger( <ClassName>.getClass )

Then the class name will be included inn the logging statement. If you ever have the pleasure of using Kibana you will understand!

Reply:
Will update